phoenix-core/src/main/java/org/apache/phoenix/compile/JoinCompiler.java

```diff
@@ -320,32 +328,57 @@ public ParseNode getPostFiltersCombined() {
             return combine(postFilters);
         }
 
-        public void addFilter(ParseNode filter) throws SQLException {
+        public void addPostJoinFilter(ParseNode parseNode) {
```

Comment:
add many methods in JoinTable for JoinCompiler.WhereNodeVisitor to make the modification of internal states of JoinTable is sealed in JoinTable class.